javascript/extractor/src/com/semmle/js/extractor/CFGExtractor.java

```diff
@@ -1583,8 +1583,16 @@ protected void visitAssign(INode assgn, INode lhs, Expression rhs) {
 
 		@Override
 		public Void visit(BinaryExpression nd, SuccessorInfo i) {
-			this.seq(nd.getLeft(), nd.getRight(), nd);
-			succ(nd, i.getGuardedSuccessors(nd));
+			if ("??".equals(nd.getOperator())) {
+				// the nullish coalescing operator is short-circuiting, but we do not add guards for it
```

Comment:
This doesn't seem to follow the same logic as `visit(LogicalExpression, SuccessorInfo)`, though? In particular, for the other short-circuiting operators we go `nd -> left -> right`, with an additional `left -> successors` edge.